docs/pages/contribution/advanced-topics/historical-state-regen.md

```diff
@@ -0,0 +1,29 @@
+---
+title: Understanding Historical Sate Regeneration
+---
+
+# Understanding Historical Sate Regeneration
+
+To run a blockchain client and establish consensus we need latest headers and forkchoice data. This operation does not require to historical data, specially after the epochs which are finalized. Storing the full state information for the finalized slots increase the storage requirement a lot and not suitable for running the node for long time.
+
+## Solution
+
+To overcome the storage problem for the archive nodes we implemented following algorithm to store and fetch the historical sates.
+
+**Approach**
+
+Assume we have following chain represents the state object every slot, with following diff layer configurations `1,2,3,5`. With assumption that we have 8 slots each epoch, The following configuration for layers implies:
+
```

Comment:
need more details here, given epoch `n` where `n % 5 = 0`, how do we store state `n, n + 1, n + 2, n + 3, n + 4`? need to coordinate with the 2 database repositories we have

also on the load size, how do we load state `n, n + 1, n + 2, n + 3, n + 4`